### PR TITLE
Adjust file realm docs

### PR DESCRIPTION
**Pull request by jkakavas**
The existing wording in the file realm docs proved confusing
for users as it seemed to indicate that it should _only_ be
used as a fallback/recovery realm and that it is not a
first class realm.

This change attempts to clarify this and point out that recovery
is _a_ use case for the file realm but not the only intended one.

### CODE REVIEW

File: x-pack/docs/en/security/authentication/file-realm.asciidoc [hunk 1]
```diff
@@ -9,15 +9,16 @@ IMPORTANT:  As the administrator of the cluster, it is your responsibility to
...
 
-The `file` realm is primarily supported to serve as a fallback/recovery realm. It
-is mostly useful in situations where all users locked themselves out of the system
-(no one remembers their username/password). In this type of scenarios, the `file`
-realm is your only way out - you can define a new `admin` user in the `file` realm
-and use it to log in and reset the credentials of all other users.
+The `file` realm is very useful as a fallback/recovery realm, for instance in cases where
+the cluster is unresponsive or the security index is unavailable, or when you forget the
+password for your administrative users.
+In this type of scenarios, the `file` realm is a convenient way out - you can
```
Comment (lcawl) [inline x-pack/docs/en/security/authentication/file-realm.asciidoc]:
```suggestion
In this type of scenario, the `file` realm is a convenient way out - you can
```

File: x-pack/docs/en/security/authentication/file-realm.asciidoc [hunk 2]
```diff
@@ -9,15 +9,16 @@ IMPORTANT:  As the administrator of the cluster, it is your responsibility to
 ensure the same users are defined on every node in the cluster. The {stack}
 {security-features} do not deliver any mechanism to guarantee this.
 
-The `file` realm is primarily supported to serve as a fallback/recovery realm. It
-is mostly useful in situations where all users locked themselves out of the system
-(no one remembers their username/password). In this type of scenarios, the `file`
-realm is your only way out - you can define a new `admin` user in the `file` realm
-and use it to log in and reset the credentials of all other users.
+The `file` realm is very useful as a fallback/recovery realm, for instance in cases where
```
Comment (lcawl) [inline x-pack/docs/en/security/authentication/file-realm.asciidoc]: Not mandatory, but I prefer spelling out "/" and splitting this long sentence in two:
```suggestion
The `file` realm is very useful as a fallback or recovery realm. For example, use it in cases where
```

**Review by lcawl — APPROVED** [review]
Minor comments, otherwise LGTM

**Review by tvernum — COMMENTED** [review]
I feel like this softens the wording too much

I think we can away people away from over using the file realm by explicitly point out the the file realm users cannot be retrieved from the API or the Kibana Security Management UI.

I can come up with some possible wording when I'm not on mobile.

**Comment by jkakavas** [comment]
> I feel like this softens the wording too much

Fair enough

> I think we can away people away from over using the file realm by explicitly point out the the file realm users cannot be retrieved from the API or the Kibana Security Management UI.

I felt like the "important" above + the docs on the user APIs are enough, but you're probably right, I'll amend this to spell it out

**Comment by jkakavas** [comment]
How about now @tvernum  ?

**Review by tvernum — APPROVED** [review]
LGTM, thanks.

**Pull request merged**